tests/console/firewalld.pm

```diff
@@ -0,0 +1,140 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test FirewallD basic usage
+# Maintainer: Alexandre Makoto Tanno <atanno@suse.com>
+
+use strict;
+use warnings;
+use base "consoletest";
+use testapi;
+use utils qw(systemctl);
+
+sub run {
+    select_console("root-console");
+
+    # Check Service State, enable it if necessary, set default zone to public
+    record_info 'Check Service State';
+    assert_script_run("if [ `systemctl is-active firewalld | grep -q -i 'inactive' ; echo $?` -eq 0 ]; then systemctl start firewalld; fi");
+    assert_script_run("firewall-cmd --set-default-zone=public");
+
+
+    # Test #1 - Stop firewalld then start it
+    record_info 'Test #1', 'Test: Stop firewalld, then start it';
+    systemctl('stop firewalld');
+    systemctl('start firewalld');
+
+    # Check Service State, enable it if necessary
+    record_info 'Check Service State';
+    assert_script_run("if [ `systemctl is-active firewalld | grep -q -i 'inactive' ; echo $?` -eq 0 ]; then systemctl start firewalld; fi");
+
+
+    # Test #2 - Temporary rules
+    record_info 'Test #2', 'Test Temporary Rules';
+    assert_script_run("firewall-cmd --zone=public --add-port=25/tcp");
+    assert_script_run("iptables -C IN_public_allow -p tcp --dport 25 -m conntrack --ctstate NEW -j ACCEPT");
+
+    assert_script_run("firewall-cmd --zone=public --add-service=ssh");
+    assert_script_run("iptables -C IN_public_allow -p tcp --dport 22 -m conntrack --ctstate NEW -j ACCEPT");
+
+    assert_script_run("firewall-cmd --zone=public --add-protocol=icmp");
+    assert_script_run("iptables -C IN_public_allow -p icmp -m conntrack --ctstate NEW -j ACCEPT");
+
+    assert_script_run("firewall-cmd --zone=public --add-port=2000-3000/udp");
+    assert_script_run("iptables -C IN_public_allow -p udp --dport 2000:3000 -m conntrack --ctstate NEW -j ACCEPT");
+
+    # Flush rules
+    record_info 'Flush Rules';
+    assert_script_run("firewall-cmd --reload");
```

Comment:
Why should --reload flush the firewall rules ?

Reply:
As these rules are temporary, --reload would "flush" them. But reading your comment, I have to agree that it looks weird and it should read "Reload default configuration".